spec/mongoid/criteria_spec.rb

```diff
@@ -3057,8 +3133,77 @@ class D
           Dictionary.all.pluck(:'description.de')
         end
 
-        it 'returns the specific translations' do
-          expect(plucked.first).to eq({'de' => 'deutsch-text'})
+        context "when legacy_pluck_distinct is set" do
+          config_override :legacy_pluck_distinct, true
+
+          it 'returns the specific translations' do
+            expect(plucked.first).to eq({'de' => 'deutsch-text'})
+          end
+        end
+
+        context "when legacy_pluck_distinct is not set" do
+          config_override :legacy_pluck_distinct, false
+
+          it 'returns the specific translations' do
+            expect(plucked.first).to eq('deutsch-text')
+          end
+        end
+      end
+
+      context 'when plucking a specific locale from _translations field' do
+
+        let(:plucked) do
+          Dictionary.all.pluck(:'description_translations.de')
+        end
+
+        context "when legacy_pluck_distinct is set" do
+          config_override :legacy_pluck_distinct, true
+
+          it 'returns the specific translations' do
+            expect(plucked.first).to eq(nil)
+          end
+        end
+
+        context "when legacy_pluck_distinct is not set" do
+          config_override :legacy_pluck_distinct, false
+
+          it 'returns the specific translations' do
+            expect(plucked.first).to eq('deutsch-text')
+          end
+        end
+      end
+
+      context 'when fallbacks are enabled with a locale list' do
+        before(:all) do
+          puts "I18n version: #{I18n::VERSION}"
+
+          require "i18n/backend/fallbacks"
+          I18n::Backend::Simple.send(:include, I18n::Backend::Fallbacks)
```

Comment:
We have evergreen configurations that have fallbacks enabled, those should be used to test fallback behavior instead of enabling callbacks in the test body.

Reply:
@p-mongo my two cents--_as a separate PR_, it would be nice if fallback tests could be incorporated into the main test suite, rather than making them a separate run config. You could probably make an RSpec helper that mocks the fallback functionality for a certain scope.

I don't see any strong argument for making I18n fallbacks require a "special" test run more so than any other feature.

Reply:
The setup in `spec/integration/i18n_fallbacks_spec.rb` loads the fallbacks module into process space. This is a global action affecting the entire test suite. To be able to test with fallbacks both on and off, there are separate configurations.

Reply:
done! (removed include fallbacks)